drivers/sx127x/sx127x_netdev.c

```diff
@@ -375,10 +375,14 @@ static int _set(netdev_t *netdev, netopt_t opt, const void *val, size_t len)
             assert(len <= sizeof(netopt_state_t));
             return _set_state(dev, *((const netopt_state_t*) val));
 
-        case NETOPT_DEVICE_MODE:
-            assert(len <= sizeof(uint8_t));
-            sx127x_set_modem(dev, *((const uint8_t*) val));
-            return sizeof(netopt_enable_t);
+        case NETOPT_DEVICE_TYPE:
+            assert(len <= sizeof(uint16_t));
+            /* Only LoRa modem is supported for the moment */
+            if (*(const uint16_t*) val == NETDEV_TYPE_LORA) {
+                sx127x_set_modem(dev, SX127X_MODEM_LORA);
+                return sizeof(uint16_t);
+            }
```

Comment:
```C
else {
    return -EINVAL;
}
```

The return-value `-ENOTSUP` means the *operation* is not supported. Here we have an unsupported value.

Reply:
Ok, I was unsure of that. Will change.

Reply:
Done